### PR TITLE
refactor: 💡 Update how WorkerDiagram is used in Target form

### PR DESCRIPTION
**Pull request by cameronperera**
## Description

:technologist: [Admin preview](https://boundary-acomi2c1y-hashicorp.vercel.app/scopes/s_1sheljmjgm/targets/t_h4m3u5vfzm)

### CODE REVIEW

File: ui/admin/app/components/form/target/details/index.hbs [hunk 1]
```diff
@@ -224,21 +224,10 @@
...
-              @ingressFilter={{this.ingressWorkerFilterEnabled}}
-            />
-          {{else}}
-            <WorkerDiagram::SingleFilter
-              @egressFilter={{this.egressWorkerFilterEnabled}}
-            />
-          {{/if}}
+          <WorkerDiagram
+            @egressFilter={{this.egressWorkerFilterEnabled}}
+            @ingressFilter={{this.ingressWorkerFilterEnabled}}
```
Comment (randallmorey) [inline ui/admin/app/components/form/target/details/index.hbs:229]: It would be nice if the argument names matched and indicated that these are booleans, e.g. `this.egressWorkerFilterEnabled`.

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/worker-diagram/dual-filter/hcp/index.hbs [hunk 2]
```diff
@@ -1,102 +1,91 @@
...
-        {{t 'resources.target.workers.filter-explainer.hcp-dual-egress-on'}}
-      </Hds::Form::HelperText>
-    </div>
-  {{else}}
-    <OrderedSeriesDiagram
-      data-test-dual-filter-hcp-egress-off-ingress-off
-      as |D|
+    </D.Group>
+  </OrderedSeriesDiagram>
+  <div class='target-worker-filter-explainer'>
```
Comment (randallmorey) [inline ui/admin/app/components/worker-diagram/dual-filter/hcp/index.hbs:21]: Can these class names be namespaces in a way that is consistent with the component?  e.g. `worker-diagram-explainer`

File: ui/admin/app/components/worker-diagram/index.hbs [hunk 3]
```diff
@@ -1,6 +1,17 @@
-{{yield
-  (hash
-    SingleFilter=(component 'worker-diagram/single-filter')
-    DualFilter=(component 'worker-diagram/dual-filter')
-  )
-}}
\ No newline at end of file
+<div class='worker-wrapper'>
```
Comment (randallmorey) [inline ui/admin/app/components/worker-diagram/index.hbs:1]: Is this necessary?  If so, why?

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/worker-diagram/index.hbs (hunk 3 above)
Comment (randallmorey) [reply ui/admin/app/components/worker-diagram/index.hbs:1]: If this is necessary, can the class name be `worker-diagram`?

File: ui/admin/app/styles/app.scss [hunk 4]
```diff
@@ -630,11 +630,15 @@
...
+  // these styles are not applied directly to the order-series-diagram
+  // because these are unique to the target form and may not be
+  // needed in other instances of the diagram
   .worker-wrapper {
-    display: flex;
-    align-items: center;
-    min-height: 11rem;
-    margin-top: sizing.rems(xs) - sizing.rems(xxxxs); // 10
+    .ordered-series-diagram {
+      align-items: center;
```
Comment (randallmorey) [inline ui/admin/app/styles/app.scss:638]: Why is this necessary?

File: ui/admin/app/styles/app.scss [hunk 5]
```diff
@@ -630,11 +630,15 @@
...
+      min-height: 11rem;
+      margin-top: sizing.rems(xs) - sizing.rems(xxxxs); // 10
```
Comment (randallmorey) [inline ui/admin/app/styles/app.scss:640]: Can we use rems directly when applied to HDS?  The diagrams are based on HDS, but the sizing helper is based on Structure.

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: ui/admin/app/styles/app.scss (hunk 4 above)
Comment (cameronperera) [reply ui/admin/app/styles/app.scss:638]: With `align-items`
Screen Shot 2023-02-08 at 12 39 31 PM
without `align-items`
Screen Shot 2023-02-08 at 12 39 42 PM

It keeps the diagram centered vertically to complement the min hight set.

File: ui/admin/app/components/worker-diagram/index.hbs (hunk 3 above)
Comment (cameronperera) [reply ui/admin/app/components/worker-diagram/index.hbs:1]: I completely forgot about the weird height issues we had and that is why the original `worker-wrapper` class was added. It has to be added back otherwise we get staggered icons. As you can see below, the HCP item is higher than the other items. Pushing a correction to my refactor with better naming for what it is used for.
Screen Shot 2023-02-08 at 12 53 14 PM

Comment (randallmorey) [inline ui/admin/app/components/worker-diagram/index.hbs]:
```suggestion
<div class='worker-diagram'>
```

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/worker-diagram/index.hbs [hunk 6]
```diff
@@ -1,6 +1,15 @@
...
+{{else if (feature-flag 'target-worker-filters-v2-ingress')}}
+  <WorkerDiagram::DualFilter
+    @egressWorkerFilterEnabled={{@egressWorkerFilterEnabled}}
+    @ingressWorkerFilterEnabled={{@ingressWorkerFilterEnabled}}
+  />
+{{else}}
+  <WorkerDiagram::SingleFilter
+    @egressWorkerFilterEnabled={{@egressWorkerFilterEnabled}}
+  />
+{{/if}}
```
Comment (randallmorey) [inline ui/admin/app/components/worker-diagram/index.hbs:15]: Minor readability update so that it's clear v2-ingress and v2-hcp can both be on at the same time.  The previous conditional could be interpreted to mean they are mutually exclusive.
```suggestion
{{#if (feature-flag 'target-worker-filters-v2-ingress')}}
  {{#if (feature-flag 'target-worker-filters-v2-hcp')}}
    <WorkerDiagram::DualFilter::Hcp
      @egressWorkerFilterEnabled={{@egressWorkerFilterEnabled}}
      @ingressWorkerFilterEnabled={{@ingressWorkerFilterEnabled}}
    />
  {{else}}
    <WorkerDiagram::DualFilter
      @egressWorkerFilterEnabled={{@egressWorkerFilterEnabled}}
      @ingressWorkerFilterEnabled={{@ingressWorkerFilterEnabled}}
    />
  {{/if}}
{{else}}
  <WorkerDiagram::SingleFilter
    @egressWorkerFilterEnabled={{@egressWorkerFilterEnabled}}
  />
{{/if}}
```

**Review by randallmorey — CHANGES_REQUESTED** [review]

**Review by DhariniJeeva — APPROVED** [review]

File: ui/admin/app/styles/app.scss (hunk 4 above)
Comment (cameronperera) [reply ui/admin/app/styles/app.scss]: Moved style changes out of this PR.

File: ui/admin/app/styles/app.scss (hunk 5 above)
Comment (cameronperera) [reply ui/admin/app/styles/app.scss]: Moved style changes out of this PR.

File: ui/admin/app/components/worker-diagram/index.hbs (hunk 3 above)
Comment (cameronperera) [reply ui/admin/app/components/worker-diagram/index.hbs]: No longer part of this PR.

File: ui/admin/app/components/worker-diagram/index.hbs (hunk 6 above)
Comment (cameronperera) [reply ui/admin/app/components/worker-diagram/index.hbs]: Updated!

File: ui/admin/app/components/worker-diagram/index.hbs (hunk 3 above)
Comment (cameronperera) [reply ui/admin/app/components/worker-diagram/index.hbs]: No longer relevant to this PR

**Review by dvonanderson — APPROVED** [review]
